lib/index.d.ts

```diff
@@ -0,0 +1,104 @@
+// Type definitions for panel 0.10.0
```

Comment:
This is great.

Although types are seen as documentation, I am not a big fan of having documentation away from code. Its a slippery slope as other devs forget to make changes to index.d.ts and things get hairy.

I have file a bug with typescript that I may end up fixing if no one fixes it. Basically we should be able to use `allowJS, checkJS and declarations:true` and automatically validate our code and generate the index.d.ts from it.

This means types are auto generated and our panel code is also type checked. We can live in js land happily ever after.

https://github.com/Microsoft/TypeScript/issues/15718

But this is a good first start.

Reply:
One year late. I was wrong 😢. Auto generating declarations is harder than I thought. Let me know what I can do to get type definitions in. Some files in panel have moved so we'll need to adjust for that.